docs/qute/TemplateSupport.md

```diff
@@ -0,0 +1,93 @@
+# Qute Template support
+
+Qute support provides `completions`, `validation`, `hover`, `definition`, `highlight`, `linked editing range` inside Qute template.
+
+To benefit with this support, we recommand to name your template file with the `*.qute.*` extension (ex: `items.qute.html`, `items.qute.txt`, `items.qute.yaml`, etc)
+
+## Parameter declaration support
+
+A `parameter declaration` is used to declare the expected model in the Qute template:
+
+```
+{@java.util.List<org.acme.qute.Item> items}
+```
+
+The Qute support helps to write easily this declaration with the completion, hover and definition support.
+
+### Completion for Java types
+
+The Qute support provides completion to write easily parameter declaration:
+
+![Completion for Java types](images/QuteTemplateParameterDeclarationCompletion.gif)
+
+### Validate Java type
+
+The Qute support validate Java types:
+
+![Validate Java type](images/QuteTemplateParameterDeclarationValidation.png)
+
+### Jump to Java type definition
+
+You can jump to the proper Java type:
+
+![Jump to Java type definition](images/QuteTemplateParameterDeclarationDefinition.gif)
+
+## Section support
+
+### Completion for Qute section
+
+Qute support provides snippet to generate quickly `Qute section`. Here a sample to generate quickly the `#for` section:

```

Comment:
provides snippets to quickly generate `Qute` sections. Here is a sample for the `#for` section:

Reply:
fixed